packages/core/src/routes/hook.ts

```diff
@@ -21,23 +24,59 @@ export default function hookRoutes<T extends AuthedRouter>(
     }
   );
 
+  router.get(
+    '/hooks/:id',
+    koaGuard({
+      params: z.object({ id: z.string() }),
+      response: Hooks.guard,
+      status: [200, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      ctx.body = await findHookById(id);
+
+      return next();
+    }
+  );
+
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, name: true, signingKey: true, enabled: true, events: true })
+        .extend({
+          // Note: ensure the user will not create a hook with an empty name.
+          name: string().min(1).optional(),
```

Comment:
should we convert this in a common place? cc @simeng-li

Reply:
min(1) automatically added for all non-null fields without a default value.  For the name case here, it has a default value.  It might be a particular case for the PATCH method not allowing users to update using an empty string. Don't have a quick answer.